### PR TITLE
Remove reflection in Nullable TryParse extension

### PR DESCRIPTION
**Pull request by cadatoiva**
This refactors the in house `T? TryParse<T>(string)` extension method to be faster by avoiding using reflection. It narrows the compatible types by doing so, but is currently only used for long and int, which are still compatible, as well as all other built in ValueTypes

### CODE REVIEW

File: src/Jackett.Common/Utils/Extensions.cs [hunk 1]
```diff
@@ -85,29 +85,24 @@ public static class KeyValuePairsExtension
...
             try
             {
-                var val = parseMethod.Invoke(null, new object[] { value });
+                var val = Convert.ChangeType(value, typeof(T));
                 return (T)val;
             }
-            catch
+            catch(Exception e) when
+                (e is InvalidCastException // Conversion not supported
+                 || e is FormatException) // Conversion failed
```
Comment (ngosang) [inline src/Jackett.Common/Utils/Extensions.cs]: Just catch ALL exceptions and return null. This method is not expected to fail.

File: src/Jackett.Common/Utils/Extensions.cs [hunk 2]
```diff
@@ -85,29 +85,24 @@ public static class KeyValuePairsExtension
 
     public static class ParseExtension
     {
-        public static T? TryParse<T>(this string value) where T : struct
+        public static T? TryParse<T>(this string value) where T : struct, IConvertible
```
Comment (ngosang) [inline src/Jackett.Common/Utils/Extensions.cs]: Could you add some unit tests for: null, empty, whitespace, "10" int, "10" double, "tt10" int .... and compare the results with the previous implementation? That is a good practice called TDD.
1. Add the unit tests first
2. The tests should pass in the old code.
3. Rewrite the code and the tests must pass in the new code.
4. All unit tests will be in place for the future changes

File: src/Jackett.Common/Utils/Extensions.cs (hunk 1 above)
Comment (cadatoiva) [reply src/Jackett.Common/Utils/Extensions.cs]: Empty catch or `catch(Exception)` are both generally considered very bad practice. At a minimum we should not be catching `ArgumentNullException` because `typeof(T)` should never return null and something has gone horribly wrong in ways that shouldn't be hidden if it does.

Should an OverflowException return null? Or should it return MaxValue or maybe a modulus of MaxValue (wrapped value)?

It is obvious that not being able to convert a string to the target value should default to null, and it is obvious that if the string is not a valid target value that it should also return null. The other two above do not have an obvious handling path, IMHO.

Comment (ngosang) [reply src/Jackett.Common/Utils/Extensions.cs]: As general rule, the methods called TrySomething() never throw an exception. They will do his best to do the job or fail in silence. Could be return null or converted value + boolean like in .Net API.
